roles/splunk/tasks/remove_splunk.yml

```diff
@@ -11,4 +11,139 @@
+      ansible.builtin.group:
+        name: "{{ splunk_nix_group }}"
+        state: absent
+        local: "{{ local_os_group }}"
+      become: true
+  when: splunk_nix_user != 'root'
+
+- name: Remove Disable-THP/Ulimit files
+  block:
+    - name: Stop Disable-THP service
+      ansible.builtin.service:
+        name: disable-thp
+        enabled: false
+        state: stopped
+      become: true
+      when:
+        - ansible_connection!="docker"
+        - "'full' in group_names"
+      ignore_errors: true
+
+    - name: Remove disable-thp service to systemd (CentOS/RHEL 7+)
+      ansible.builtin.file:
+        path: /etc/systemd/system/disable-thp.service
+        state: absent
+      become: true
+      notify:
+        - reload systemctl daemon
+      when:
+        - ansible_os_family == "RedHat"
+        - ansible_distribution_major_version|int > 6
+
+    - name: Remove ulimits
+      ansible.builtin.file:
+        path: /etc/security/limits.d/splunk_ulimits.conf
+        state: absent
+      become: true
+      when: splunk_use_initd
+
+- name: Remove Dmesg configs
+  block:
+    - name: Add sysctl for dmesg
+      ansible.posix.sysctl:
+        name: kernel.dmesg_restrict
+        value: '0'
+        state: absent
+        sysctl_file: /etc/sysctl.d/20-splunk.conf
+        reload: true
+      register: sysctl_dmesg
+      become: true
+
+    - name: Restart procps
+      ansible.builtin.service:
+        name: procps
+        state: restarted
+      when: sysctl_dmesg.changed
+      become: true
+  when: configure_dmesg
+
+- name: Remove firewall artifacts
+  block:
+    - name: Remove firewalld artifacts
+      block:
+        - name: Remove splunk firewalld service
+          ansible.builtin.file:
+            path: /etc/firewalld/services/{{ splunk_firewall_service }}.xml
+            state: absent
+          become: true
+
+        - name: Remove firewalld service
+          ansible.posix.firewalld:
+            service: "{{ splunk_firewall_service }}"
+            permanent: true
+            state: enabled
+            immediate: true
+          notify: reload firewalld
+          become: true
+      when: firewall_service == "firewalld"
+
+    - name: Remove UFW ports
+      community.general.ufw:
+        port: "{{ item.number }}"
+        proto: "{{ item.protocol }}"
+        rule: allow
+        state: reloaded
+        comment: "{{ item.desc | default('') }}"
+        delete: true
+      become: true
+      loop: "{{ splunk_firewall_ports }}"
+      when: firewall_service == "ufw"
+  when:
+    - firewall_service != 'undefined'
+    - configure_firewall != false
+    - "'full' in group_names"
+
+- name: Remove FACLs
+  block:
+    - name: Add setfacl to logrotate script
+      ansible.builtin.lineinfile:
+        path: /etc/logrotate.d/syslog
+        insertbefore: '  endscript'
+        line: '        /usr/bin/setfacl -Rm u:{{ splunk_nix_user }}:rx /var/log'
+        state: absent
+      become: true
```

Comment:
NB Likely gonna be updated in https://github.com/splunk/ansible-role-for-splunk/pull/160/files#diff-032c335b5fa8c728153c887ba6051c71b11474b8ba98682616b9ac79d2605968

Reply:
Let me know when it's merged - definitely better to have it as a var, that should be a PR all of it's own, to do:

- Ulimit file
- THP Service file
- DMesg file
- Auditd conf file